src/auth/auth.service.ts

```diff
@@ -33,73 +37,66 @@ export class AuthService {
     return accessToken;
   }
 
-  async login(phoneNumber, response) {
+  async login(phoneNumber: string) {
     const { phoneNumber: phoneNumberWithoutCountryCode, countryCode } =
       getCountryCodeAndPhoneNumber(phoneNumber);
+
     const user = await this.userService.findOrCreateUser(
       phoneNumberWithoutCountryCode,
       countryCode,
     );
     const token = this.getAuthToken(user);
-    return response.json({
+
+    return {
       token,
       user,
-    });
-  }
-
-  async verifyFirebaseOtp(sessionInfo, code) {
-    const apiKey = process.env.FIREBASE_WEB_API_KEY;
-    const url = `https://identitytoolkit.googleapis.com/v1/accounts:signInWithPhoneNumber?key=${apiKey}`;
-    const requestBody = {
-      sessionInfo,
-      code,
     };
+  }
 
+  async verifyFirebaseOtp(payload: string) {
     try {
-      const response = await fetch(url, {
-        method: 'POST',
-        headers: {
-          'Content-Type': 'application/json',
-        },
-        body: JSON.stringify(requestBody),
-      });
-
-      const responseData = await response.json();
-      return responseData;
+      if (!app) {
+        app = firebase.initializeApp({
```

Comment:
Initialization of the app will happen every time you call the app
It can be shifted to the class level using constructor

Reply:
Yeah...I sort of forced functional programming paradigm onto this 😅 . If you check line 10 on this file, the `app` variable is declared above.

Also, the firebase app should only be initialised in staging and production, not in dev env.

Probably have to have a separate dev stage/prod otp verification service like in frontend.

Reply:
Moved the firebase app init code to a utils file